sql/rpl_gtid.cc

```diff
@@ -3361,13 +3364,28 @@ static my_bool report_audit_findings(void *entry, void *report_ctx_arg)
   return FALSE;
 }
 
-my_bool Binlog_gtid_state_validator::report(FILE *out, my_bool is_strict_mode)
+my_bool Binlog_gtid_state_validator::report(FILE *out, my_bool is_strict_mode,
+                                            my_bool report_out_of_order,
+                                            rpl_gtid *stop_gtids, size_t n_stop_gtids)
 {
-  struct gtid_report_ctx report_ctx;
-  report_ctx.out_file= out;
-  report_ctx.is_strict_mode= is_strict_mode;
-  report_ctx.contains_err= FALSE;
+  struct gtid_report_ctx report_ctx= {out, is_strict_mode, report_out_of_order, FALSE};
   my_hash_iterate(&m_audit_elem_domain_lookup, report_audit_findings, &report_ctx);
+  // Because we don't want to add a new struct gtid_report_ctx member just to index the stop GTIDs,
+  // we can't check them in the report_audit_findings iteration and have to look them up one by one.
+  for (size_t i= 0; i < n_stop_gtids; ++i)
+  {
+    rpl_gtid *stop_gtid= &stop_gtids[i];
```

Comment:
It would be cleaner if you could move this behind a method like `is_expected_gtid()` and then the condition at 3382 could be:
```
if (is_expected_gtid(stop_gtid[i])) ...
```
In general, I like the code to read as "what is done" and not "how it's done", unless I want to drop lower in the abstraction.

Reply:
I looked around but found that all other instances with similar search and compare has different actions for “domain not found” and “sequence unexpected”.
Therefore, unless we replace all those actions with lambdas (our C++ standard just happens to have those), I don’t find it enough value to add a single-use function, especially that the condition is already readable (“drop lower in the abstraction”) before or after [`@bnestere`’s inversion](#discussion_r1824594249).